security/check_for_log4j_attacks.md

```diff
@@ -0,0 +1,41 @@
+# Check for CVE-2021-44228 Log4j attacks in request header
+
+Validate all parts of the request header against the attack and
+also check for evasions of the detection.
+
+```lua
+-- detect CVE-2021-44228 Log4j
+local function check_for_attack (location, value)
+    lower = string.lower(value)
+    if string.contains(lower, "${jndi:") then
+      avi.vs.log("CVE-2021-44228 Log4j attack detected at " .. location)
+      avi.http.response(400)
+    end
+    -- detect evasion
+    if string.contains(lower, "${${") or string.contains(lower, "${j${") or string.contains(lower, "${jn${") or string.contains(lower, "${jnd${") or string.contains(lower, "${jndi${") then
+          avi.vs.log("CVE-2021-44228 Log4j attack (evasion attempt) detected at " .. location)
+          avi.http.response(400)
+    end
+end
+
+local function check_table(location, table)
+    if table then
+        for name, value in pairs(table) do
+           check_for_attack(location, name)
+           if type(value) == "table" then
+               for index, value in pairs(value) do
+                   check_for_attack(location .. "." .. name, value)
+               end
+           else
+               check_for_attack(location .. "." .. name, value)
+           end
+        end
+    end
+end
+
```

Comment:
Mirko As discussed, we can buffer request body with DS API's and then subject to string search. We have 3 choices

- Use WAF with our CRS rules 
- Use Datascript with potential lua code for url decoding. 
- Use Datascript and check for patterns (including URL encoded ones)
-